src/pages/DataLoader/DataLoader.tsx

```diff
@@ -0,0 +1,321 @@
+        />
+      </div>
+    );
+  };
+
+  const validateFileName = () => {
+    let valid = true;
+    // access-policy.json would cause the backend to break
+    if (fileName.includes('access-policy')) {
+      setMessage('File name cannot include access-policy');
+      valid = false;
+    }
+    if (!fileName) {
+      setMessage('File name cannot be empty');
+      valid = false;
+    }
+    return valid;
+  };
+
+  const renderButton = () => {
+    const buttonData = {
+      validate: {
+        method: validate,
+        text: 'Validate',
+      },
+      upload: {
+        method: upload,
+        text: 'Upload',
+      },
+      harvest: {
+        method: harvest,
+        text: 'Load',
+      },
+    };
+
+    return (
+      <Button disabled={running} onClick={buttonData[state].method}>
+        {running ? 'Running' : buttonData[state].text}
+      </Button>
+    );
+  };
+
+  const validate = async () => {
+    if (!file) {
+      setMessage('Please select a file before running validation');
+      return;
+    }
+
+    setRunning(true);
+    setMessage('Validating file');
+    try {
+      if (fileType === 'access-policy') validateAccessPolicy();
+      if (fileType === 'stac') await validateSTAC();
+      setMessage('File successfully validated');
+      setState('upload');
+    } catch (error) {
+      console.error(error);
+    }
+    setRunning(false);
+  };
+
+  const validateAccessPolicy = async () => {
+    if (!file) {
+      setMessage('Please select a file before running validation');
+      return;
+    }
+
+    setRunning(true);
+    setMessage('Validating Access Policy');
+    try {
+      const text = await file.text();
+      JSON.parse(text);
+      setMessage('File successfully validated');
+      setState('upload');
+      setFileName('access-policy.json');
+    } catch {
+      setMessage('Invalid JSON');
+    }
+    setRunning(false);
+  };
+
+  const validateSTAC = async () => {
+    const stacContent = await file.text();
+    let stac;
+    try {
+      stac = JSON.parse(stacContent);
+    } catch (e) {
+      setMessage('❌ Invalid JSON format');
+      throw new Error();
+    }
+
+    let data;
+
+    try {
+      const res = await fetch(`https://dev.eodatahub.org.uk/api/validate-stac`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          credentials: 'include',
+        },
+        body: JSON.stringify({
+          content: stac,
+        }),
+      });
+      const dataAndCode = await res.json();
+      data = dataAndCode[0];
+    } catch (error) {
+      setMessage('Failed to send request');
+      throw new Error();
+    }
+
+    if (data.status === 'error') {
+      if (!data.content) {
+        setMessage('❌ Failed to validate STAC');
+        setValidationErrors([data.message]);
+        throw new Error();
+      }
+      const errors = [];
+      data.content.forEach((error) => {
+        if (Array.isArray(error)) {
```

Comment:
This isn't necessarily true, if I look at the response I see:
```
        "content": [
            [
                "Item name '20201211_223832_CS2' should only contain Searchable identifiers",
                "Identifiers should consist of only lowercase characters, numbers, '_', and '-'"
            ],
            [
                "A STAC collection should contain a summaries field",
                "It is recommended to store information like eo:bands in summaries"
            ]
        ]
```

Where the nested arrays are actually related, one is the problem, the other is more of a description.
This should be made evident in the UI as well, as it looks like there are 4 errors, when there are actually only 2